Dart/src/com/jetbrains/lang/dart/ide/documentation/DartDocumentationProvider.java

```diff
@@ -94,18 +98,89 @@ else if (componentType == DartComponentType.CLASS) {
     return resultUrl.toString();
   }
 
-  @Override
-  public String generateDoc(PsiElement element, PsiElement originalElement) {
-    return DartDocUtil.generateDoc(element);
+  @Nullable
+  private static String generateDocServer(PsiElement element) {
```

Comment:
Idea: unify doc preview regardless of the source of information: server or IDE. Extract a method that will build resulting HTML based on prepared input data.